### PR TITLE
New package: AMD.AMDSoftware version 23.7.1

### PR DESCRIPTION
**Pull request by RobertHarnett**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/112415&drop=dogfoodAlpha

### CODE REVIEW

**Comment by RobertHarnett** [comment]
I tried to source the package by using [the official link from the AMD driver's page](https://www.amd.com/en/support). However, as you can see from the [failed download page](https://www.amd.com/en/support/kb/faq/download-incomplete) the only way the download starts is if in the http request you send a 'Referrer' tag of `https://www.amd.com/`. 

Is this possible to do with winget? I was able to do it with just the command line but unable to figure it out with Winget.
`iwr https://drivers.amd.com/drivers/installer/23.10/whql/amd-software-adrenalin-edition-23.7.1-minimalsetup-230706_web.exe -Headers @{Referer="https://www.amd.com/"}`

As a result of this, the actual source of the download is from www.techpowerup.com, which is a reputable source, but a third-party one, nonetheless.

I also suspect since it's GPU drivers, and requires a reboot before use, the installer validation won't work automatically, I have tested it on my own machine (with AMD GPU), and it installs correctly with expected behaviour.

File: manifests/a/AMD/AMDSoftware/23.7.1/AMD.AMDSoftware.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+PackageIdentifier: AMD.AMDSoftware
+PackageVersion: 23.7.1
+Installers:
+- Architecture: x64
+  InstallerType: exe
+  InstallerUrl: https://nl2-dl.techpowerup.com/files/5jJKL2mGgFCciEu_xReQWg/1689732063/whql-amd-software-adrenalin-edition-23.7.1-win10-win11-july6.exe
+  InstallerSha256: 8D9C54320A7A7D791F7BE73D98CEE7C9A4C70B553CC7E945803E61BEC388168F
+  InstallerSwitches:
+    Silent: -INSTALL
+    SilentWithProgress: -INSTALL
```
Comment (Trenly) [inline manifests/a/AMD/AMDSoftware/23.7.1/AMD.AMDSoftware.installer.yaml:13]:
```suggestion
    SilentWithProgress: -INSTALL
  ExpectedReturnCodes: 
  - InstallerReturnCode: 173
    ReturnResponse: systemNotSupported
```

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by ItzLevvie** [comment]
[Policy] Hardware

**Pull request closed without merging**